samples_project/Assets/SampleViewer/Samples/FeatureLayer/FeatureData.cs

```diff
@@ -0,0 +1,66 @@
+using Esri.ArcGISMapsSDK.Components;
+using Esri.GameEngine.Geometry;
+using Esri.HPFramework;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(ArcGISLocationComponent))]
+public class FeatureData : MonoBehaviour
+{
+    private HPTransform cameraHP;
+    private ArcGISLocationComponent cameraLocationComponent;
+    private float distance;
+    private HPTransform featureHP;
+    private ArcGISLocationComponent locationComponent;
+    private double RayCastDistanceThreshold = 300000;
+    private double scale;
+    private double SpawnHeight = 10000;
+    
+    public ArcGISCameraComponent ArcGISCamera;
+    public List<double> Coordinates = new List<double>();
+    public Renderer FeatureRender;
+    public List<string> Properties = new List<string>();
+
+    private void Start()
+    {
+        cameraLocationComponent = ArcGISCamera.GetComponent<ArcGISLocationComponent>();
+        featureHP = transform.GetComponent<HPTransform>();
+        cameraHP = ArcGISCamera.GetComponent<HPTransform>();
+        featureHP = transform.GetComponent<HPTransform>();
+        InvokeRepeating("DynamicScale", 2.0f, 0.5f);
+        InvokeRepeating("SetOnGround", 0.1f, 0.3f);
+    }
+
+    private void DynamicScale()
+    {
+        scale = cameraLocationComponent.Position.Z * 25.0 / 20000.0f;
```

Comment:
I really don't like the magic numbers here. Also why not just use a single value?
```suggestion
        scale = cameraLocationComponent.Position.Z * 0.00125f;
```